Src/BootCamp.Chapter/ContactsCenter.cs

```diff
@@ -1,66 +1,108 @@
 ﻿using System;
 using System.Collections.Generic;
+using System.Globalization;
 using System.IO;
-using System.Text;
 
 namespace BootCamp.Chapter
 {
     public class ContactsCenter
     {
-         private readonly string _peopleFile;
+        private readonly string _peopleFile;
 
         public ContactsCenter(string peopleFile)
         {
             if (!File.Exists(peopleFile))
             {
-                throw new FileNotFoundException("File is made up"); 
+                throw new FileNotFoundException("File is made up");
             }
-                      
-            _peopleFile = File.ReadAllText(peopleFile); 
-           
+
+            _peopleFile = File.ReadAllText(peopleFile);
+
             if (String.IsNullOrWhiteSpace(_peopleFile))
             {
-                throw new Exception(); 
+                throw new InvalidOperationException("File cannot be empty");
             }
-         
         }
 
-       
         /// <summary>
         /// Gets people by filter predicate.
         /// </summary>
         /// <returns></returns>
         public List<Person> Filter(Predicate<Person> predicate)
         {
             var people = new List<Person>();
-            var splittedList = _peopleFile.Split($"{Environment.NewLine}");
-            for (int i = 1; i < splittedList.Length -1 ; i++)
+            var splittedList = _peopleFile.Split(Environment.NewLine);
+            // skipped the first row because its a header.
+            for (int i = 1; i < splittedList.Length - 1; i++)
             {
-                var person = ConvertStringToPerson(splittedList[i]);
+                var person = TryParse(splittedList[i]);
                 if (predicate(person))
                 {
                     people.Add(person);
                 }
-
             }
-                   
+
             return people;
         }
 
-        public Person ConvertStringToPerson(string personString)
+        public Person TryParse(string personString)
         {
-            var splittedPerson = personString.Split(',');
+            if (String.IsNullOrEmpty(personString))
+            {
+                throw new ArgumentException("String cannot be empty");
+            }
+
             var person = new Person();
+            CultureInfo culture = new CultureInfo("en-US");
+            var splittedPerson = personString.Split(',');
+
+            if (String.IsNullOrEmpty(splittedPerson[0]))
+            {
+                throw new ArgumentException("name cannot be empty");
+            }
+
+            if (String.IsNullOrEmpty(splittedPerson[1]))
+            {
+                throw new ArgumentException("surename cannot be empty");
```

Comment:
All of this validation should be a part of person constructor, if it really matters. I thin you are being too strict here and most of this is not really needed.

Reply:
oke, I will delete all the valiidation.